src/auth/index.js

```diff
@@ -26,12 +37,33 @@ export default class Auth {
       storageType: 'localStorage'
     }, options);
 
-    this.client = client;
-    this.rootUrl = rootUrl;
-    this.codec = options.codec;
-    this.platform = options.platform || _platform;
-    this.storage = createStorage(options);
-    this.providers = createProviders(this, options);
+    auth.client = client;
+    auth.rootUrl = rootUrl;
+    auth.codec = options.codec;
+    auth.platform = options.platform || _platform;
+    auth.storage = createStorage(options);
+    auth.providers = createProviders(auth, options);
+
+    Promise.all([
```

Comment:
From the previous round of comments but looks like it was not implemented: this should be what is returned instead of doing `return new Promise((resolve, reject) => {` and resolving with `auth`. You can do the `return Promise.all().then(... return auth)`

Reply:
👍